components/brave_wallet/common/fil_address.cc

```diff
@@ -0,0 +1,163 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/fil_address.h"
+#include <charconv>
+
+#include "base/check_op.h"
+#include "base/logging.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "brave/third_party/argon2/src/src/blake2/blake2.h"
+#include "components/base32/base32.h"
+
+namespace brave_wallet {
+
+namespace {
+
+#define SECP256K_PUBLIC_KEY_SIZE 20
+#define SECP256K_ADDRESS_SIZE 41
+#define BLS_ADDRESS_SIZE 86
+#define BLS_PUBLIC_KEY_SIZE 48
+
+std::vector<uint8_t> BlakeHash(const std::vector<uint8_t>& payload,
+                               size_t length) {
+  blake2b_state blakeState;
+  if (blake2b_init(&blakeState, length) != 0) {
+    VLOG(0) << __func__ << ": blake2b_init failed";
+    return std::vector<uint8_t>();
+  }
+  if (blake2b_update(&blakeState, payload.data(), payload.size()) != 0) {
+    VLOG(0) << __func__ << ": blake2b_update failed";
+    return std::vector<uint8_t>();
+  }
+  std::vector<uint8_t> result;
+  result.resize(length);
+  if (blake2b_final(&blakeState, result.data(), length) != 0) {
+    VLOG(0) << __func__ << ": blake2b_final failed";
+    return result;
```

Comment:
We should use `std::vector<uint8_t>()` explicitly.
Actually maybe it's easier in the first place to return absl::optional<std::vector<uint8_t>> for this  function, and we just return absl::nullopt for all error cases.

Reply:
done